hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -163,7 +163,7 @@ def queries(query_group,
 
         cmd = [__grains__['osquerybinpath'], '--read_max', MAX_FILE_SIZE, '--json', query_sql]
         t0 = time.time()
-        res = __salt__['cmd.run_all'](cmd)
+        res = __salt__['cmd.run_all'](cmd, timeout=90)
```

Comment:
Let's change this to 10800 seconds (3 hours). We have some queries that run for a ridiculously long time, but do eventually finish. We want to continue to debug those queries, so we want them to finish and report their query run length into Splunk. We'll track them down and try to fix them.

This fix is just to kill queries that have become indefinitely hung.

Reply:
Additionally, is there a way to know when the command ended due to going over the timeout (and log that)? That would give us indication when a query ran but took longer than our allowed time.

Reply:
Yeah, when the timeout hits, it logs the stderr and some other things. It wouldn't be super obvious, but if you went looking for missing osquery results, you'd see a message about it and the stderr for the killed proc.

Changing to 10ksec on the nose. I like ksec as a time unit. I can make it precisely 3 hours if you need. Please, ksec!!